src/tools/perf/lib/libperf.c

```diff
@@ -1299,8 +1299,16 @@ ucx_perf_do_warmup(ucx_perf_context_t *perf, const ucx_perf_params_t *params)
         /* Stop when reaching the deadline */
         stop_status = (ucs_get_time() > deadline) ? UCS_OK : UCS_INPROGRESS;
 
-        /* Synchronize on whether to continue or stop the warmup phase */
-        status = ucx_perf_test_exchange_status(perf, stop_status);
+        if (params->thread_count == 1) {
+            status = ucx_perf_test_exchange_status(perf, stop_status);
+        } else {
+#pragma omp barrier
+#pragma omp single copyprivate(status)
+            /* Synchronize on whether to continue or stop the warmup phase */
+            status = ucx_perf_test_exchange_status(perf, stop_status);
+#pragma omp barrier
```

Comment:
does it mean that multiple threads are always built with OMP?

Reply:
yes, when perftest is running multiple parallel threads for the same "perf" context, it's always by OpenMP